src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -30,6 +30,45 @@ internal void VerifyDiagnostics(string source, CSharpCompilationOptions options,
     public class LocalFunctionTests : LocalFunctionsTestBase
     {
         [Fact]
+        public void TypeParameterBindingScope()
+        {
+            var src = @"
+class C
+{
+    public void M()
+    {
+        {
+            int T = 0; // Should not have error
+
+            int Local<T>() => 0; // Should conflict with above
+            Local<int>();
+            T++;
+        }
+        {
+            int T<T>() => 0;
```

Comment:
>int T<T>() => 0 [](start = 12, length = 15)

It would be good to have a test when type parameter ```T``` conflicts with is a type parameter of a containing method/local function.